File: 66.plus-one.c
/*
 * @lc app=leetcode id=66 lang=c
 *
 * [66] Plus One
 */


/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* plusOne(int* digits, int digitsSize, int* returnSize){
    int carry = 1;
    for(int i = digitsSize-1; i >= 0;i--){
        digits[i]=(digits[i]+carry)%10;
        carry=(digits[i]+carry)/10;
    }
    *returnSize = digitsSize;

    int* returnDigits;
    // if (carry){
    //     *returnSize++;
    //     returnDigits = malloc(sizeof(int)*(*returnSize));
    //     returnDigits[0]=1;          
    // }else{
        returnDigits = malloc(sizeof(int)*(*returnSize));
    // }
    for(int i = 0 ;i<digitsSize;i++){
        returnDigits[i+1]=digits[i];
    } 
    return returnDigits;
}

File: 66.plus-one.py
#
# @lc app=leetcode id=66 lang=python3
#
# [66] Plus One
#

# @lc code=start
from typing import *
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        digits.reverse()
        res = []
        carry = 1
        for digit in digits:
            res.append((digit+carry)%10)
            carry = (digit + carry) // 10
        if carry:
            res.append(carry)
        res.reverse()
        return res 
# @lc code=end

if __name__ == "__main__":
    print(Solution().plusOne([9]))
